arrow-avro/src/reader/record.rs

```diff
@@ -219,6 +220,10 @@ enum Decoder {
     Boolean(BooleanBufferBuilder),
     Int32(Vec<i32>),
     Int64(Vec<i64>),
+    DurationSecond(Vec<i64>),
+    DurationMillisecond(Vec<i64>),
+    DurationMicrosecond(Vec<i64>),
+    DurationNanosecond(Vec<i64>),
```

Comment:
```suggestion
    #[cfg(feature = "avro_custom_types")]
    DurationSecond(Vec<i64>),
    #[cfg(feature = "avro_custom_types")]
    DurationMillisecond(Vec<i64>),
    #[cfg(feature = "avro_custom_types")]
    DurationMicrosecond(Vec<i64>),
    #[cfg(feature = "avro_custom_types")]
    DurationNanosecond(Vec<i64>),
```

If the feature flag is off, we'd just read it as an `Int64`